railties/lib/rails/application/default_middleware_stack.rb

```diff
@@ -34,7 +34,7 @@ def build_stack
           # Must come after Rack::MethodOverride to properly log overridden methods
           middleware.use ::Rails::Rack::Logger, config.log_tags
           middleware.use ::ActionDispatch::ShowExceptions, show_exceptions_app
-          middleware.use ::ActionDispatch::DebugExceptions, app
+          middleware.use ::ActionDispatch::DebugExceptions, app, config.api_only
```

Comment:
@matthewd agreed about the new config. That could also be done for the session as well, etc.